service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -193,6 +202,85 @@ public void addDirectionsToCaseDetails(CaseDetails caseDetails,
         });
     }
 
+    @SuppressWarnings("unchecked")
+    public List<Element<Direction>> getDirectionsForUnrepresentedParties(CaseDetails caseDetails,
+                                                                         List<Element<Direction>> directions) {
+        final boolean direction = true;
+
+        return directions.stream()
+            .filter(element -> {
+                    List<Element<UUID>> ids = new ArrayList<>();
+
+                    List<Element<Representative>> representatives = defaultIfNull(
+                        (List<Element<Representative>>) caseDetails.getData().get("representatives"), emptyList());
+
+                    // if assigned to specific person and representatives exist -> find representative ids
+                    // if not -> return direction.
+                    if (directionIsValidForFiltering(element, representatives)) {
```

Comment:
I think we could add another filter step to just filter directionsValidForFiltering. We would have one less if in that lambda.

Reply:
extracted